http/src/request/channel/reaction/create_reaction.rs

```diff
@@ -50,31 +49,32 @@ impl<'a> CreateReaction<'a> {
         Self {
             channel_id,
             emoji,
-            fut: None,
             http,
             message_id,
         }
     }
 
-    fn request(&self) -> Request {
-        Request::from_route(Route::CreateReaction {
-            channel_id: self.channel_id.0,
-            emoji: self.emoji.display().to_string(),
-            message_id: self.message_id.0,
-        })
+    fn request(self) -> (Request, &'a Client) {
```

Comment:
The `request` method consumes the token, and so it needs to consume `self`. The HTTP client is returned out of it. I've found some instances where tokens are cloned instead of consumed so I've updated them to the same pattern and added some documentation.